crates/fluvio-controlplane-metadata/src/mirror/spec.rs

```diff
@@ -75,4 +74,25 @@ pub struct Home {
     pub id: String,
     pub remote_id: String,
     pub public_endpoint: String,
+    #[cfg_attr(feature = "use_serde", serde(skip_serializing_if = "Option::is_none"))]
+    pub tls: Option<ClientTls>,
```

Comment:
TLS should be mandatory

Reply:
@sehz we had this conversation on Friday. TLS should not be mandatory, as there are 3 use cases.

Reply:
It always been mandatory to have TLS for these connections:
- Edge to Home
- Home

Edge doesn't not need to have TLS

Reply:
I understand that run a mirroring in production without TLS is super risky, but maybe the user wants to test it locally first without certificate overhead.

Maybe we should just print warns that this is not a secure connection?

Reply:
As we discussed in Friday, when I test this locally, I don't need TLS connection.

Reply:
I don't get it. 

no certs = no authentication
with certs = we have authentication

This is already working like it.

Do you mean create something to encrypt it even without authentication?

Reply:
authentication is mandatory as without it there is no way to ensure data integrity which lead to corruption.

Reply:
This is not going to be solved here. Let's keep optional until we discuss.

Reply:
So there are two different questions:
(1) What strategy for Authentication (TLS, None, etc).
(2) Whether this flag is needed which will be depend on answering first question.  If this is needed, name should be change to so intent is clear.

Reply:
Decision:  Edge authentication  is  optional for mirroring,.  However TLS is mandatory if auth is required

Reply:
But in this case, TLS should be attached to Edge not to home.

Reply:
Edge cluster is storing TLS in record/struct called Home